inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -4248,4 +4328,34 @@ public void close() {
             }
         }
     }
+
+    private class DefaultBeanResolutionContext extends AbstractBeanResolutionContext {
```

Comment:
Rename to `SingletonBeanResolutionContext` since there is already a `DefaultBeanResolutionContext` in `io.micronaut.context`